xtime/range_test.go

```diff
@@ -0,0 +1,99 @@
+package xtime_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/birdie-ai/golibs/xtime"
+	"github.com/google/go-cmp/cmp"
+)
+
+// tm returns a time with the given hour and minute, and other fields hard-coded.
+func tm(hour, minute int) time.Time {
+	return time.Date(2023, 1, 1, hour, minute, 0, 0, time.UTC)
+}
+
+func TestRangeContains(t *testing.T) {
+	cases := []struct {
+		from, to, t time.Time
+		want        bool
+	}{
+		{tm(1, 0), tm(2, 0), tm(0, 59), false},
+		{tm(1, 0), tm(2, 0), tm(1, 0), true},
+		{tm(1, 0), tm(2, 0), tm(1, 1), true},
+		{tm(1, 0), tm(2, 0), tm(1, 59), true},
+		{tm(1, 0), tm(2, 0), tm(2, 0), false},
+		{tm(1, 0), tm(2, 0), tm(2, 1), false},
+	}
+	for _, c := range cases {
+		got := xtime.Range{c.from, c.to}.Contains(c.t)
+		if got != c.want {
+			t.Errorf("xtime.Range{%v, %v}.Contains(%v) == %v, want %v",
+				c.from, c.to, c.t, got, c.want)
+		}
+	}
+}
+
+func TestRangeDuration(t *testing.T) {
+	cases := []struct {
+		from, to time.Time
+		want     time.Duration
+	}{
+		{tm(1, 0), tm(1, 0), time.Duration(0)},
+		{tm(1, 0), tm(2, 0), 1 * time.Hour},
+		{tm(1, 0), tm(3, 0), 2 * time.Hour},
+	}
+	for _, c := range cases {
+		got := xtime.Range{c.from, c.to}.Duration()
+		if got != c.want {
+			t.Errorf("xtime.Range{%v, %v}.Duration() == %v, want %v",
+				c.from, c.to, got, c.want)
+		}
+	}
+}
+
+func TestRangeSplit(t *testing.T) {
+	cases := []struct {
+		from, to time.Time
+		max      time.Duration
+		want     []xtime.Range
+	}{
+		{
+			tm(1, 0),
+			tm(2, 0),
+			2 * time.Hour,
+			[]xtime.Range{{tm(1, 0), tm(2, 0)}},
+		},
+		{
+			tm(1, 0),
+			tm(2, 0),
+			1 * time.Hour,
```

Comment:
```suggestion
			time.Hour,
```

You probably just like this....so ignore me being annoying :rofl:

Reply:
I find it a bit more readable -- in English you'd say "it takes one hour" not "it takes hour". Maybe that's just me though.

Reply:
I imagined you had some reasoning like that :laughing: ...I find it odd because you are saying more than it is necessary, it feels like I need to spend some extra cycles to be sure that isn't something else going on =P (because it is a no-op/identity operation.. so my mind always go to "why ?" instead of just reading as english).

Reply:
OK, if others find it confusing then I'll change it

Reply:
it could be just me :laughing: ...feel free to ignore it

Reply:
I'm quite ok with 1 as well actually :)

Reply:
See, it is just me being an awkward motherfucker :rofl: